lifting-state-up/single-source-of-truth.md

```diff
@@ -0,0 +1,260 @@
+
+While we're here, we're also going to add `id` data to each student. This helps the data feel a little more realistic, as if we're reading it from a database or API.
+
+<!-- prettier-ignore-start -->
+```js
+function App() {
+  const studentData = [
+    {
+      id: 1,
+      nameData: 'Ada',
+      emailData: 'ada@dev.org',
+      isPresentData: false
+    },
+    {
+      id: 2,
+      nameData: 'Soo-ah',
+      emailData: 'sooah@dev.org',
+      isPresentData: false
+    },
+    {
+      id: 3,
+      nameData: 'Chrissy',
+      emailData: 'chrissy@dev.org',
+      isPresentData: true
+    }
+  ];
+
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <StudentList students={studentData}></StudentList>
+    </main>
+  );
+}
+```
+<!-- prettier-ignore-end -->
+
+In this example, we've chosen the name `isPresentData` to help us clarify that this is coming from the student dataset. We can change the name to anything else if we want to!
+
+### Update PropTypes
+
+Now that `id` and `isPresentData` are in our `studentData`, and we pass that data into `StudentList` and `Student`, we should update our PropTypes.
+
+The PropTypes for `StudentList` can look like this:
+
+<!-- prettier-ignore-start -->
+```js
+StudentList.propTypes = {
+    students: PropTypes.arrayOf(PropTypes.shape({
+        id: PropTypes.number.isRequired,
+        nameData: PropTypes.string.isRequired,
+        emailData: PropTypes.string.isRequired,
+        isPresentData: PropTypes.bool
+    }))
+};
+```
+<!-- prettier-ignore-end -->
+
+And for `Student`:
+
+<!-- prettier-ignore-start -->
+```js
+Student.propTypes = {
+    id: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    email: PropTypes.string.isRequired,
+    isPresent: PropTypes.bool
+};
+```
+<!-- prettier-ignore-end -->
+
+### Ensure `StudentList` Passes the New `App` Data
+
+We introduced the new pieces of data, `id` and `isPresentData`, in `App`. This data gets sent to `StudentList` through the prop named `students`.
+
+Let's ensure that this data gets used!
+
+`StudentList` should read this data, and then sends it to `Student` components.
+
+<!-- prettier-ignore-start -->
+```js
+const StudentList = (props) => {
+    const studentComponents = props.students.map((student, index) => {
+        return (
+            <li key={index}>
+                <Student
+                    id={student.id}
+                    name={student.nameData}
+                    email={student.emailData}
+                    isPresent={student.isPresentData}
+                ></Student>
+            </li>
+        );
+    });
+
+    // ... return some JSX
+}
+```
+<!-- prettier-ignore-end -->
+
+<!-- available callout types: info, success, warning, danger, secondary  -->
+### !callout-info
+
+## Refactoring Opportunity: `key=student.id`
+
+In a later commit, we could update our `map` call so that the `key` pulls the ID from the student data, not from the list index.
```

Comment:
Sofia
```suggestion
In a later commit, Sofia could update the `map` call so that the `key` pulls the ID from the student data, not from the list index.
```